### DISCUSSION TITLE
Use async/await in examples as often as possible?

### DISCUSSION THREAD

**Discussion by greggman** [mdn/mdn-community#565] [Content] [answered]
MDN URL: to many to list

#### What information was incorrect, unhelpful, or incomplete?

I'm curious in general if there is a policy on async/await vs then/catch?

#### What did you expect to see?

I expect to see pretty much every MDN article the uses async features (fetch, service workers, etc...) to show mainly or even only async/await based code. For example I expect this

```
 async function main() {
    const response = await fetch('https://example.com');
    const data = await response.json();
    // use data
}
```

not this

```
fetch('https://example.com/)
   .then(response => response.json())
   .then(data => {
      // use data
   });
```

To me the first is simple and easy to understand for new users. The second is effectively obfusticated code.

Does the leadership of MDN have some policy on this issue? Is it okay to edit articles using obfusticated `then` code to start using straight forward `async`/`await` code?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/service_worker_api/using_service_workers`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/service_worker_api/using_service_workers/index.html
* Last commit: https://github.com/mdn/content/commit/f6f11d2a249520f01bad93d7c4a5e5cdf6ed914a
* Document last modified: 2021-02-19T19:33:30.000Z

</details>

**Comment by CreaTorAlexander** [comment]
I agree to use async / await consistently.

**Comment by jpmedley** [comment]
They tend to shy away from new idioms until they've had to achieve support in most browsers. When to embrace them is a judgement call with no reliable heuristic.

**Comment by wbamberg** [comment, marked as answer]
We have a policy on this! https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/JavaScript#use_modern_js_features which says it's fine to use async/await. It doesn't actually _encourage_ it (although personally I think it should).

But I think it would be fine to update the example you've quoted.

**Comment by pepelsbey** [comment]
Closing as answered/settled.
